nix/shared.nix

```diff
@@ -1,7 +1,22 @@
-{ lib, ... }: {
+{ config, lib, ... }: {
   options = {
-    programs.nix-index-database.comma.enable = lib.mkEnableOption "wrapping comma with nix-index-database and put it in the PATH";
+    programs = {
+      nix-index-database.comma.enable = lib.mkEnableOption "wrapping comma with nix-index-database and put it in the PATH";
+
+      # TODO: Remove this option after the release of nixos-25.11.
+      nix-index.acknowledgeBreakingChange = lib.mkOption {
```

Comment:
Do we even need this option? Can we not just not have not a default set?

Reply:
This option was introduced in commit https://github.com/nix-community/nix-index-database/pull/132/commits/e35a2c27f8ebf0f4bb5e9bd46e394c2e9c56bd84 to address the following concern:

> Should we maybe start by printing a warning if the option is set to true by this module, to give people the chance to update their configs?
>
> -- https://github.com/nix-community/nix-index-database/pull/132#issuecomment-2539645646

However, I am fine with reverting this commit.

Reply:
I think not providing a default at all is indeed better, it will print a clear error that most nix users will easily understand.

Including this option is also a future maintenance issue, we need to decide when to remove it (which will be again a breaking change for people who decided to set it), and it also doesn't make it clear which breaking change you're acknowledging, in case we introduce a new one before we remove this option.

So, I'd remove this option and just not provide a default for the enable option for now.

Reply:
Alright. I dropped commit 395c618edf86 ("shared: add programs.nix-index.acknowledgeBreakingChange option") in [v9](https://github.com/nix-community/nix-index-database/pull/132#issuecomment-2842053846), if that is what you meant.